yarn-project/aztec-nr/aztec/src/log.nr

```diff
@@ -14,9 +14,25 @@ fn emit_encrypted_log<N>(
 }
 
 fn emit_unencrypted_log<T>(
+    context: &mut PublicContext,
+    log: T,
+) {
+    let contract_address = context.this_address();
+    let event_selector = 5; // TODO: compute actual event selector.
+    let _ = oracle::logs::emit_unencrypted_log(contract_address, event_selector, log);
+    // context.accumulate_unencrypted_logs(log);
+}
+
+// TODO: We might want to remove this since emitting unencrypted logs from private functions is violating privacy.
+// --> might be a better approach to force devs to make a public function call that emits the log if needed then
+// it would be less easy to accidentally leak information.
+// If we decide to keep this function around would make sense to wait for traits and then merge it with emit_unencrypted_log.
+fn emit_unencrypted_log_from_private<T>(
     context: &mut PrivateContext,
     log: T,
 ) {
-    let _ = oracle::logs::emit_unencrypted_log(log);
-    context.accumulate_unencrypted_logs(log);
+    let contract_address = context.this_address();
```

Comment:
Could we make the `contract_address` 0 here as a potential fix for leaking of the contract address?

Reply:
We will use this address to verify in the kernel circuit that the log was emitted from the correct contract. If we didn't do this verification then a malicious contract could emit a log on behalf of another one. So having it 0 for private logs is not an option once the private logs are fully generalized.

Currently the private logs are only used to emit encrypted note preimages and there we actually don't need to do this contract address verification because we check that the note is committed to in the private data tree (better name would be note hash tree).